employees/functional_tests/employees_tests.py

```diff
@@ -0,0 +1,256 @@
+        self.assertEqual(self.__get_number_of_employees_on_current_page(), 3)
+
+    def __check_employee_list_and_return_boolean_status(self, list_given_by_tester):
+        elements = self.selenium.find_elements_by_name('employee-from-list')
+        employees_list = []
+
+        for element in elements:
+            employees_list.append(element.text)
+
+        for name in list_given_by_tester:
+            if name not in employees_list:
+                return False
+        return True
+
+    def __get_number_of_employees_on_current_page(self):
+        elements = self.selenium.find_elements_by_name('employee-from-list')
+        return len(elements)
+
+    def __check_if_employee_filter_exist_and_get_boolean_value(self, filter_name_enum):
+
+        filter_name = filter_name_enum.value
+        try:
+            element = self.selenium.find_element_by_id(filter_name)
+
+        except:
+            element = None
+
+        self.assertNotEqual(element, None, 'unable to find ' + filter_name)
+
+        return element.is_selected()
+
+    def __check_if_pagination_block_doesnt_exist(self):
+
+        def __find_paginate_previous_radio():
+            self.selenium.find_element_by_id('paginated')
+
+        self.assertRaises(exceptions.NoSuchElementException, __find_paginate_previous_radio)
+
+    def __check_if_previous_page_exist_and_is_disabled(self):
+        try:
+            element = self.selenium.find_element_by_xpath('//label[@class="my-button previous disabled"]')
+        except:
+            element = None
+        self.assertNotEqual(element, None, 'unable to find previous page label with status disabled')
+
+    def __check_if_previous_page_exist_and_is_clickable(self):
+        try:
+            element = self.selenium.find_element_by_xpath('//label[@class="my-button previous clickable"]')
+        except:
+            element = None
+        self.assertNotEqual(element, None, 'unable to find previous page label with status clickable')
+
+    def __check_if_next_page_exist_and_is_disabled(self):
+        try:
+            element = self.selenium.find_element_by_xpath('//label[@class="my-button next disabled"]')
+        except:
+            element = None
+        self.assertNotEqual(element, None, 'unable to find next page label with status disabled')
+
+    def __check_if_next_page_exist_and_is_clickable(self):
+        try:
+            element = self.selenium.find_element_by_xpath('//label[@class="my-button next clickable"]')
+        except:
+            element = None
+        self.assertNotEqual(element, None, 'unable to find next page label with status clickable')
+
+    def __check_page_exist_and_get_true_if_selected(self, number):
+        try:
+            id = 'pg-label' + str(number) + '-checked'
+            self.selenium.find_element_by_id(id)  # test if element exist. If doesn't throws exceptions
+            return True
+        except:
+            try:
```

Comment:
This is strait abomination, the whole test need to be rethink and rewritten. Use try/except as little as possible. And never one inside another.

Reply:
is it ok for simple method checking if element exist. I want to add custom comment to exception:

    def __check_if_previous_page_exist_and_is_clickable(self):
        try:
            self.selenium.find_element_by_xpath('//label[@class="my-button previous clickable"]')
        except NoSuchElementException:
            raise NoSuchElementException('unable to find previous page label with status clickable')